src/s2_cells/mod.rs

```diff
@@ -247,10 +250,14 @@ impl S2Cells {
             .collect();
         let mut cell_union = CellUnion(point_cells);
         cell_union.normalize();
-        let convex_hull = cell_union.rect_bound();
+        let rect = cell_union.rect_bound();
+        self.cells_intersecting(|cell: &Cell| rect.intersects_cell(cell))
+    }
+
+    fn cells_intersecting(&self, f: impl Fn(&Cell) -> bool) -> Vec<S2CellId> {
```

Comment:
Would it be a bit more elegant to just take a s2::Region?

Reply:
Good idea, I changed it.